lsp/server/src/validateHtml.ts

```diff
@@ -0,0 +1,48 @@
+import { HTMLDocument, getLanguageService } from 'vscode-html-languageservice';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { Diagnostic } from 'vscode-languageserver/node';
+import { DiagnosticProducer } from './diagnostic/DiagnosticProducer';
+import {
+    DiagnosticSettings,
+    isTheDiagnosticSuppressed
+} from './diagnostic/DiagnosticSettings';
+import { MobileOfflineFriendly } from './diagnostic/html/mobileOfflineFriendly';
+
+const diagnosticProducers: DiagnosticProducer<HTMLDocument>[] = [
+    new MobileOfflineFriendly()
+];
+
+function parseHTMLContent(content: TextDocument): HTMLDocument {
+    const htmlLanguageService = getLanguageService();
+    return htmlLanguageService.parseHTMLDocument(content);
+}
+
+export async function validateHtml(
+    setting: DiagnosticSettings,
+    textDocument: TextDocument
+): Promise<Diagnostic[]> {
+    let results: Diagnostic[] = [];
+
+    const producers = diagnosticProducers.filter((producer) => {
+        return !isTheDiagnosticSuppressed(setting, producer.getId());
+    });
+
+    if (producers.length > 0) {
+        try {
```

Comment:
Today such try catch block too big issue is noticed.  It swallows the exception from the parseHtml call exception and calls to await producer.validateDocument.   If there are more than one producers, and if the first producer throws, the second producer will be not be called ... 
We have similar issue in validateJs and valideGraphql.  Time to abstract and centralize?

Reply:
I followed your code pattern. And this is one of the reasons why we shouldn't prescribe to unproven pattern or scaffolding because it's likely that these antipatterns emerge.

Your suggestion is valid though. But it shouldn't be done for this PR.